### PR TITLE
[Harmony] Adding get validators and get delegations

### PR DESCRIPTION
**Pull request by gupadhyaya**

### CODE REVIEW

File: platform/harmony/client.go [hunk 1]
```diff
@@ -41,6 +43,32 @@ func (c *Client) GetBlockByNumber(num int64) (info BlockInfo, err error) {
...
+	return
+}
+
+func (c *Client) GetBalance(address string) (string, error) {
+	var result string
+	err := c.RpcCall(&result, "hmy_getBalance", []interface{}{address, "latest"})
+	if err != nil {
+		return "0", err
+	}
+	bln, err := numbers.HexToDecimal(result)
```
Comment (vikmeup) [inline platform/harmony/client.go]:
```suggestion
	balance, err := numbers.HexToDecimal(result)
```

File: platform/harmony/client.go [hunk 2]
```diff
@@ -41,6 +43,32 @@ func (c *Client) GetBlockByNumber(num int64) (info BlockInfo, err error) {
...
+	if err != nil {
+		return "0", err
+	}
+	return bln, nil
```
Comment (vikmeup) [inline platform/harmony/client.go]:
```suggestion
	return balance, nil
```

File: platform/harmony/stake.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	}
+	return results
+}
+
+func getDetails() blockatlas.StakingDetails {
+	return blockatlas.StakingDetails{
+		Reward:        blockatlas.StakingReward{Annual: Annual},
+		MinimumAmount: blockatlas.Amount("0"),
+		LockTime:      0,
+		Type:          blockatlas.DelegationTypeAuto,
```
Comment (vikmeup) [inline platform/harmony/stake.go]: `DelegationTypeAuto ` - this means tokens staked automatically, like Algorand

Use `delegate` type

File: platform/harmony/api.go [hunk 4]
```diff
@@ -13,6 +13,8 @@ type Platform struct {
 	client Client
 }
 
+const Annual = 10
```
Comment (vikmeup) [inline platform/harmony/api.go]: we need to be dynamic to avoid changing

Comment (gupadhyaya) [reply platform/harmony/api.go]: changed it to rely on rpc call which will be implemented.

File: platform/harmony/stake.go [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	delegations, err := p.client.GetDelegations(address)
+	if err != nil {
+		return nil, err
+	}
+
+	validators, err := services.GetValidatorsMap(p)
+	if err != nil {
+		return nil, err
+	}
+	results = append(results, NormalizeDelegations(delegations.List, validators)...)
```
Comment (Pantani) [inline platform/harmony/stake.go]: why did you create an empty result slice only to append the normalized delegations?

Comment (gupadhyaya) [reply platform/harmony/stake.go]: my bad. removed.

File: platform/harmony/stake_test.go [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+var validator1 = blockatlas.StakeValidator{
+	ID:     "one1pdv9lrdwl0rg5vglh4xtyrv3wjk3wsqket7zxy",
+	Status: true,
+	Info: blockatlas.StakeValidatorInfo{
+		Name:        "",
+		Description: "",
+		Image:       "",
+		Website:     "",
```
Comment (Pantani) [inline platform/harmony/stake_test.go]: can you cover the stake validator info as well?

**Comment by Pantani** [comment]
@gupadhyaya can you update your branch from master?

**Comment by vikmeup** [comment]
Re-open pull request when ready

**Comment by EnoRage** [comment]
Moved to https://github.com/trustwallet/blockatlas/pull/923https://github.com/trustwallet/blockatlas/pull/923

**Pull request closed without merging**